Document JavaFlowSupport (#28315)
akka/akka: akka-docs/src/main/paradox/stream/reactive-streams-interop.md

<a id="reactive-streams-integration"></a>
## Overview

Akka Streams implements the [Reactive Streams](http://reactive-streams.org/) standard for asynchronous stream processing with non-blocking
back pressure. 

> Since Java 9 the APIs of Reactive Streams has been included in the Java Standard library, under the  `java.util.concurrent.Flow` 

Reviewer: ```suggestion
Since Java 9 the Reactive Streams APIs are included in the Java Standard library, under the  `java.util.concurrent.Flow` 
```

namespace. For Java 8 there is instead a separate Reactive Streams artifact with the same APIs in the package `org.reactivestreams`.

Akka streams provides interoperability for both these two API versions, the Reactive Streams interfaces directly through factories on the

> regular `Source` and `Sink` APIs. For the Java 9 and later built in interfaces there is a separate set of factories in @apidoc[JavaFlowSupport].

Reviewer: ```suggestion
Akka Streams provides interoperability for both APIs:
- For the Reactive Streams interfaces through factories on the regular `Source` and `Sink` APIs. 
- For the built-in interfaces in Java 9 and later via factories in @apidoc[JavaFlowSupport].
```

> In the following samples the standalone Reactive Stream API factories has been used but each such call can be replaced with the

Reviewer: ```suggestion
The following samples use the standalone Reactive Streams factories but each such call can be replaced with the
```

corresponding method from `JavaFlowSupport` and the JDK @scala[`java.util.concurrent.Flow._`]@java[`java.util.concurrent.Flow.*`] interfaces.

> Note that it is not possible to use `JavaFlowSupport` on Java 8 since the needed interfaces simply is not available in the Java standard library.

Reviewer: ```suggestion
Note that it is not possible to use `JavaFlowSupport` on Java 8 since the needed interfaces are not available in the Java standard library.
```
